src/credentials/FabricTable.cpp

```diff
@@ -438,16 +438,13 @@ CHIP_ERROR FabricInfo::VerifyCredentials(const ByteSpan & noc, const ByteSpan &
 
 FabricTable::~FabricTable()
 {
-    FabricTableDelegate * delegate = mDelegate;
+    FabricTable::Delegate * delegate = mDelegateListRoot;
     while (delegate)
     {
-        FabricTableDelegate * temp = delegate->mNext;
-        if (delegate->mOwnedByFabricTable)
-        {
-            chip::Platform::Delete(delegate);
-        }
-        delegate = temp;
+        FabricTable::Delegate * temp = delegate->next;
+        delegate                     = temp;
     }
+    mDelegateListRoot = nullptr;
```

Comment:
Why bother, in a destructor?  Are we worrying about later use-after-free of the fabric table?